oeps/oep-0005.rst

```diff
@@ -0,0 +1,187 @@
+   hurts any development that it doing file-heavy operations (such as manipulating
+   static assets).
+3. The monolithic `Vagrant`_ image isn't particularly representative of the
+   production service architecture, because all of the services are installed
+   on the same machine, which makes it harder for developers to be confident
+   that their changes will work when deployed.
+4. The shared services dependencies on things like MySQL and Elasticsearch mean
+   that all of the services in Devstack have to be upgraded at the same time,
+   or have to maintain compatibility with multiple database versions until all
+   services are ready to update. This hinders independent development on the
+   services.
+5. Because all of the services are installed on the same image, the image either
+   has to be updated by every user after downloading (to update to the latest
+   code versions), or the image has to be rebuilt every time any of the installed
+   services change (which results in lots of churn).
+6. Because the `Vagrant`_ image contains many separate services, the definition
+   of that image has to be separate from any of the service repositories, which
+   makes it hard to update (and to push updates out to users).
+
+
+Specification
+=============
+
+Interface
+~~~~~~~~~
+
+The primary developer interaction with ``devstack`` will be through ``make``
+shortcuts. The following targets will be standard::
+
+    make devstack.watch  # Start the devstack services (if they aren't already)
+                         # and display their logs on stdout
+
+    make devstack.shell  # Start the devstack services (if they aren't already)
+                         # and access a shell on the primary container
+                         # (as the user that owns the service, in the directory
+                         # that has the service source code)
+
+    make devstack.clean  # Stop the devstack services and remove any intermediate
+                         # data that they stored in the .devstack/ directory
+
+    make devstack.*      # Start the devstack services (if they aren't already)
+                         # and run `make *` from the code directory of the service
+                         # (as the service owner)
+
+Implementation
+~~~~~~~~~~~~~~
+
+Requirements
+------------
+
+1. Services that aren't part of the Open edX ecosystem must be run using
+   their own published containers (where available).
+2. Open edX containers must run all of their processes via `supervisor`_ to
+   provide a single entrypoint even when the container has multiple processes
+   (such as `nginx`_ and a `Django`_ service running under `gunicorn`_).
+3. Open edX services running in containers must log to ``stdout`` and ``stderr``
+   so that those logs can be captured by `supervisor`_.
+4. Dependent services (such as databases) should be configured to mount their
+   storage directories in ``.devstack/`` inside the source code directory (this
```

Comment:
This feels unnecessarily prescriptive.  I think that that the convention for all persistent stores should be defined under a base directory, that directory should have a default location, that location should be configurable.  Personally, I'd like that location to be outside of the git working tree and personally I don't like defaulting to a hidden directory.